controlplane/kubeadm/main.go

```diff
@@ -209,6 +209,7 @@ func main() {
 					&corev1.ConfigMap{},
 					&corev1.Secret{},
 				},
+				Unstructured: true,
```

Comment:
This is slightly different from what we are doing for other containers, which is to have an uncached client and use it surgically... 
What about keeping it consistent with other providers so we can easily spot we are doing cached reads for unstructured while we gain some more confidence?

Reply:
I couldn't find any other places where we use unstructured.Unstructured in KCP. So it seemed reasonabe to change it globally

Reply:
Added a comment